server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -172,4 +173,11 @@ public final String simpleName() {
      *  Both {@code this} and {@code mergeWith} will be left unmodified. */
     public abstract Mapper merge(Mapper mergeWith);
 
+    /**
+     * Called when this {@linkplain Mapper} is parsed on the {@code _search}
+     * request to make fields available just for this search. 
+     */
+    public List<MappedFieldType> convertToSearchTimeMappings() {
```

Comment:
could this rather be a boolean isRuntimeField ? I see though that you are also resolving a Mapper into a list of MappedFieldTypes, but why a list? are you thinking of grok that will expand to multiple fields?

Reply:
I had it this way for grok and for the other things I was experimenting with. And because `Mapper` doesn't have a MappedFieldType on it, only FieldMapper. But I think changing how we parse will make this less of a thing. If you'd prefer to start with a boolean I'm happy to do that and change it for grok if we need to.

Reply:
yes please I would prefer not to worry about grok in this PR. the cast Mapper to FieldMapper is odd indeed, maybe we can address that upstream.